### ISSUE TITLE
Are conv_variational layers trainable with TF 2 and Keras-style fitting?

### ISSUE THREAD

**Issue by skeydan** [tensorflow/probability#567]
Referring to

 https://groups.google.com/a/tensorflow.org/d/msg/tfprobability/axQpJqs0oHc/XSCCVKseEQAJ

and the example I gave there 

```
model = tf.keras.Sequential([
tf.keras.layers.Reshape([32, 32, 3]),
tfp.layers.Convolution2DReparameterization(
64, kernel_size=5, padding='SAME', activation=tf.nn.relu),
tf.keras.layers.MaxPooling2D(pool_size=[2, 2],
strides=[2, 2],
padding='SAME'),
tf.keras.layers.Flatten(),
tfp.layers.DenseReparameterization(10),
])

# compile
# fit

# after training, model.losses has the right length but the first tensor is "empty":

[[1]]
Tensor("sequential/conv3d_flipout/divergence_kernel:0", shape=(), dtype=float32)

[[2]]
tf.Tensor(51614.953, shape=(), dtype=float32)
```

could someone please provide insight on why there does not seem to be a KL loss when training with Keras? Are these layers supposed to be trained with Keras?
(Using TF 2.)

Many thanks in advance!!

**Comment by alexv1247** [comment]
I have a similar issue being that a model trained with the keras fit function and tensorflow probability layers is not training at all. Furthermore, since I have two output losses, the overall model loss seems to have no link to the two output losses. The overall loss is 45000 and the two output losses are each 1,3

`      
       import numpy as np
       from tensorflow import keras
       import tensorflow_probability as tfp
       import tensorflow as tf
       from plot.plot_utils import plot_model_metrics
       from Custom_Keras_layers.ProbSqueezeExcite import squeeze_excite_block

    def build(self, squeeze_excite, dropout_rate, lr):

        inp = keras.layers.Input(shape=[self.timesteps, self.features])
        # left side
        # 1 Conv1D block
        l = tfp.layers.Convolution1DFlipout(filters=2*self.features, kernel_size=2, padding='same', activation=tf.nn.relu)(inp)
        l = keras.layers.BatchNormalization()(l)
        if squeeze_excite == 1:
            l = squeeze_excite_block(l)
        l = keras.layers.Dropout(dropout_rate)(l, training=True)

        # 1 Conv1D block
        l = tfp.layers.Convolution1DFlipout(filters=4 * self.features, kernel_size=4, padding='same', activation=tf.nn.relu)(l)
        l = keras.layers.BatchNormalization()(l)
        if squeeze_excite == 1:
            l = squeeze_excite_block(l)
        l = keras.layers.Dropout(dropout_rate)(l, training=True)

        # 1 lstm bock
        l = keras.layers.LSTM(32, recurrent_dropout=dropout_rate, dropout=dropout_rate)(l, training=True)

        # letf output layer
        l = tfp.layers.DenseFlipout(self.classes, activation=tf.nn.softmax, name='left')(l)

        # right side
        # 1 Conv1D block
        r = tfp.layers.Convolution1DFlipout(filters=2 * self.features, kernel_size=2, padding='same', activation=tf.nn.relu)(inp)
        r = keras.layers.BatchNormalization()(r)
        if squeeze_excite == 1:
            r = squeeze_excite_block(r)
        r = keras.layers.Dropout(dropout_rate)(r, training=True)

        # 1 Conv1D block
        r = tfp.layers.Convolution1DFlipout(filters=4 * self.features, kernel_size=4, padding='same', activation=tf.nn.relu)(r)
        r = keras.layers.BatchNormalization()(r)
        if squeeze_excite == 1:
            r = squeeze_excite_block(r)
        r = keras.layers.Dropout(dropout_rate)(r, training=True)

        # 1 lstm bock
        r = keras.layers.LSTM(32, recurrent_dropout=dropout_rate, dropout=dropout_rate)(r, training=True)

        # letf output layer
        r = tfp.layers.DenseFlipout(self.classes, activation=tf.nn.softmax, name='right')(r)

        model = keras.models.Model(inputs=inp, outputs=[l, r])

        optimizer = tf.train.AdamOptimizer(learning_rate=lr)
        losses = {
            "left": self._neg_log_likelihood_bayesian,
            "right": self._neg_log_likelihood_bayesian}
        model.compile(optimizer=optimizer, loss=losses, metrics=['accuracy'])
        self.model = model

    def _neg_log_likelihood_bayesian(self, y_true, y_pred):
        neg_log_likelihood = tf.nn.softmax_cross_entropy_with_logits_v2(labels=tf.argmax(y_true, axis=-1), logits=y_pred)
        kl = sum(self.model.losses)
        loss = neg_log_likelihood + kl
        return loss`

The model always just returns 0.25 as prediction accuracy, since it always predicts just the same class
